python/addons/embed_bert_pytorch.py

```diff
@@ -70,6 +82,263 @@ def run(self, tokens, vocab):
+
+
+@register_vectorizer(name="dict-wordpiece1d-with-heads-dict")
+class DictWordPieceVectoizer1DWithHeadsDict(WordPieceVectorizer1DWithHeadsDict):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.fields = listify(kwargs.get('fields', 'text'))
+        self.delim = kwargs.get('token_delim', '~~')
+
+    def iterable(self, tokens):
+        return super().iterable(_token_iterator(self, tokens))
+
+
+@register_vectorizer(name='heads-wordpiece1d')
+class HeadedWordPieceVectorizer1D(WordPieceVectorizer1D):
+    """Output a list of indices that are the heads of the subword tokens."""
+
+    def iterable(self, tokens):
+        j = 1  # Don't point at [CLS]
+        for tok in tokens:
+            # Unk is something we want to label
+            if tok == '<unk>':
+                yield j
+            # We don't want to label [SEP]
+            elif tok == '<EOS>':
+                pass
+            else:
+                for i, subtok in enumerate(self.tokenizer.tokenize(tok)):
+                    if i == 0:
+                        yield j  # Head
+                    else:
+                        j += 1
+            j += 1
+
+    def _next_element(self, tokens, vocab):
+        for atom in self.iterable(tokens):
+            yield atom
+
+    def run(self, tokens, vocab):
+        if self.mxlen < 0:
+            self.mxlen = self.max_seen
+        vec1d = np.zeros(self.mxlen, dtype=np.long)
+        for i, atom in enumerate(self._next_element(tokens, vocab)):
+            if i == self.mxlen:
+                i -= 1
+                break
+            vec1d[i] = atom
+        valid_length = i + 1
+        return vec1d, valid_length
+
+    def get_dims(self):
+        return self.mxlen,
+
+    def count(self, tokens):
+        seen = 0
+        counter = collections.Counter()
+        # Count the mxlen of the actual text otherwise we are to short
+        # If this was a mask we wouldn't have this issue
+        for tok in super().iterable(tokens):
+            counter[tok] += 1
+            seen += 1
+        self.max_seen = max(self.max_seen, seen)
+        return counter
+
+
+@register_vectorizer(name='dict-heads-wordpiece1d')
+class DictHeadedWordPieceVectorizer1D(HeadedWordPieceVectorizer1D):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.fields = listify(kwargs.get('fields', 'text'))
+        self.delim = kwargs.get('token_delim', '@@')
+
+    def iterable(self, tokens):
+        return super().iterable(_token_iterator(self, tokens))
+
+    def count(self, tokens):
+        return super().count(_token_iterator(self, tokens))
+
+
+class LabelHeadedWordPieceVectorizer1D(DictHeadedWordPieceVectorizer1D):
+    """Convert labels so just the heads have labels, other labels are PAD"""
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.fields = listify(kwargs.get('fields', ['text', 'y']))
+        self.delim = kwargs.get('token_delim', '~~')
```

Comment:
AFAIK this is one of the first vectorizers that actually use multiple fields from the conll file at the same time. I had to change the token delim from `@@` because there was a `@` as a token. This resulted in token that looked like `@@@O` and a split of `[''. '@O']` the empty token caused problems because the token seqs for this vectorizer wasn't aligned with the others